File: FlutterProjects/Practice/hr_and_pms - v5 Upgrading/lib/features/leave/model/LeaveType.dart
enum LeaveType { SICK, UNPAID, RESERVE }

extension LeaveTypeExtension on LeaveType {
  static LeaveType fromString(String leaveTypeString) {
    return LeaveType.values.firstWhere(
          (leaveType) => leaveType.toString().split('.').
          last.toUpperCase() == leaveTypeString.toUpperCase(),
    );
  }

  String toShortString() {
    return toString().split('.').last.toUpperCase();
  }
}

File: FlutterProjects/Practice/Backend with JWT- v8 Upgrading/src/main/java/com/mrahmed/HRandPayrollManagementSystem/entity/LeaveType.java
package com.mrahmed.HRandPayrollManagementSystem.entity;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum LeaveType {
    SICK,
    UNPAID,
    RESERVE;

    @JsonCreator
    public static LeaveType fromString(String value) {
        return LeaveType.valueOf(value.toUpperCase());
    }

    @JsonValue
    public String toJson() {
        return this.name();
    }

}

